node/src/accountant/big_int_db_processor.rs

```diff
@@ -20,67 +19,120 @@ use std::marker::PhantomData;
+                config.key_param_value(),
                 e
             ))),
         }
     }
+}
+
+impl<T: TableNameDAO + 'static> Default for BigIntDbProcessor<T> {
+    fn default() -> BigIntDbProcessor<T> {
+        Self {
+            overflow_handler: Box::new(UpdateOverflowHandlerReal::default()),
+            phantom: Default::default(),
+        }
+    }
+}
 
-    fn update_threatened_by_overflow<'a>(
+impl<T: TableNameDAO> BigIntDbProcessor<T> {
+    fn prepare_statement<'a>(
+        form_of_conn: Either<&'a dyn ConnectionWrapper, &'a Transaction>,
+        sql: &'a str,
+    ) -> Statement<'a> {
+        match form_of_conn {
+            Either::Left(conn) => conn.prepare(sql),
+            Either::Right(tx) => tx.prepare(sql),
+        }
+        .expect("internal rusqlite error")
+    }
+}
+
+pub trait UpdateOverflowHandler<T>: Debug + Send {
+    fn update_with_overflow<'a>(
+        &self,
+        conn: Either<&dyn ConnectionWrapper, &Transaction>,
+        config: BigIntSqlConfig<'a, T>,
+    ) -> Result<(), BigIntDbError>;
+}
+
+#[derive(Debug)]
+struct UpdateOverflowHandlerReal<T: TableNameDAO> {
+    phantom: PhantomData<T>,
+}
+
+impl<T: TableNameDAO> Default for UpdateOverflowHandlerReal<T> {
+    fn default() -> Self {
+        Self {
+            phantom: Default::default(),
+        }
+    }
+}
+
+impl<T: TableNameDAO> UpdateOverflowHandler<T> for UpdateOverflowHandlerReal<T> {
+    fn update_with_overflow<'a>(
         &self,
         conn: Either<&dyn ConnectionWrapper, &Transaction>,
         config: BigIntSqlConfig<'a, T>,
     ) -> Result<(), BigIntDbError> {
         let select_sql = config.select_sql();
-        let mut select_stm = Self::prepare_statement(conn, &select_sql);
+        let mut select_stm = BigIntDbProcessor::<T>::prepare_statement(conn, &select_sql);
         match select_stm.query_row([], |row| {
-            let low_bytes_result = row.get::<usize, i64>(0);
-            match low_bytes_result {
-                Ok(low_bytes) => {
-                    let wei_change_params = &config.params.wei_change_params;
-                    let high_bytes_correction = wei_change_params[0].1 + 1;
-                    let low_bytes_correction = ((low_bytes as i128
-                        + wei_change_params[1].1 as i128)
-                        & 0x7FFFFFFFFFFFFFFF) as i64
-                        - low_bytes;
-                    let update_sql = config.prepare_update_sql();
-                    let mut update_stm = Self::prepare_statement(conn, &update_sql);
+            let high_bytes_result = row.get::<usize, i64>(0);
+            let low_bytes_result = row.get::<usize, i64>(1);
+            match [high_bytes_result, low_bytes_result] {
+                [Ok(previous_high_bytes), Ok(previous_low_bytes)] => {
+                    let requested_wei_change = &config.params.wei_change_params;
+                    let (high_bytes_correction, low_bytes_correction) = Self::correct_bytes(
```

Comment:
"Correction" implies that this is a factor that can be combined with the wrong value to correct into the right value. But this is the complete correct value; perhaps it could be named something like `xxx_bytes_corrected`?